src/container/BooksForm.js

```diff
@@ -8,31 +13,63 @@ const bookCategories = [
   'Sci-Fi',
 ];
 
-// To use useState whenI have to create a new category because setState is currently redundant
-
-const formOpt = category => (
-  <option value={category} key={Math.random()}>{category}</option>
-);
-
-const BooksForm = () => (
-  <>
-    <h2>Add a Book</h2>
-    <form>
-      <div className="form-group">
-        <input type="text" placeholder="Enter book title" />
-      </div>
-      <div className="form-group">
-        <select>
-          {bookCategories.map(cat => (
-            formOpt(cat)
-          ))}
-        </select>
-      </div>
-      <div className="actions">
-        <button type="submit">Add Book</button>
-      </div>
-    </form>
-  </>
-);
-
-export default BooksForm;
+const BooksForm = ({ createBook }) => {
+  const [state, setState] = useState({
+    id: 3,
+    title: '',
+    category: '',
+  });
+
+  const handleChange = e => {
+    e.preventDefault();
+    setState({
+      id: state.id,
+      title: e.target.tagName === 'INPUT' ? e.target.value : state.title,
+      category: e.target.tagName === 'SELECT' ? e.target.options[e.target.selectedIndex].value : state.category,
+    });
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    if (state.title !== '' && state.category !== '') {
+      createBook(state);
+      setState({
+        id: state.id + 1,
+        title: '',
+        category: '',
+      });
```

Comment:
- [ ] Let's reset the state so that we reset the form after a successful submit.
![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)